rcl/src/rcl/security_directory.c

```diff
@@ -125,25 +125,25 @@ char * exact_match_lookup(
   const char * node_name,
   const char * node_namespace,
   const char * ros_secure_root_env,
-  const rcl_allocator_t * allocator)
+  const rcl_allocator_t allocator)
```

Comment:
I read some of rcutils code and it seems like rcutils is not passing the allocator using a pointer. Hence the change here. I can change it back, up to you. :)

Reply:
I see, I don't know then, let's let the final OSRF reviewer decide in this case. Keeping this open for now.

Reply:
Since the change is orthogonal to fixing overflows and leaks please keep them in a separate PR to make it easier to review the important part of the patch.

Reply:
@dirk-thomas sure.